rascal-lsp/src/main/rascal/lang/rascal/lsp/refactor/Rename.rsc

```diff
@@ -625,45 +643,74 @@ set[Define] getCursorDefinitions(list[Tree] cursor, Tree(loc) getTree, TModel(Tr
     return {};
 }
 
-tuple[set[loc], set[loc]] findOccurrenceFiles(set[Define] defs, list[Tree] cursor, Tree(loc) getTree, Renamer r) {
-    set[loc] getSourceFiles() {
-        return {*find(srcFolder, "rsc")
-            | wsFolder <- r.getConfig().workspaceFolders
-            , srcFolder <- r.getConfig().getPathConfig(wsFolder).srcs
-        };
-    }
+private set[loc] getSourceFiles(Renamer r) {
+    c = r.getConfig();
+    return {*find(srcFolder, "rsc")
+        | wsFolder <- c.workspaceFolders
+        , srcFolder <- c.getPathConfig(wsFolder).srcs
+    };
+}
+
+private bool isLocal(IdRole role) = role in variableRoles && role !:= moduleVariableId();
+
+tuple[set[loc], set[loc], set[loc]] findOccurrenceFiles(set[Define] defs, list[Tree] cursor, str newName, Tree(loc) getTree, Renamer r) {
+    if ({IdRole role} := defs.idRole) {
+        <t, desc> = asType(role);
+        if (tryParseAs(t, rascalEscapeName(newName)) is nothing) {
+            r.error(cursor[0], "\'<newName>\' is not a valid <desc>");
+            return <{}, {}, {}>;
+        }
+
+        name = "<cursor[0]>";
+        sourceFiles = getSourceFiles(r);
+        set[loc] defFiles = {};
+        set[loc] useFiles = {};
+        set[loc] newNameFiles = {};
+
+        if (role notin {variableId(), patternVariableId(), moduleId()}) {
```

Comment:
should maybe be just like the `asType`  function? so we can say `if (canInfluenceOtherFiles(role))` or something?

Reply:
Solved by implementing some pattern-matching overloads for `findOccurrenceFilesUnchecked`